inference-engine/src/readers/ir_reader/ie_ir_parser.cpp

```diff
@@ -984,7 +984,6 @@ template <>
 std::shared_ptr<ngraph::Node> V10Parser::LayerCreator<ngraph::op::v0::LSTMCell>::createLayer(
     const ngraph::OutputVector& inputs, const pugi::xml_node& node, const Blob::CPtr& weights,
     const GenericLayerParams& layerParsePrms) {
-    checkParameters(inputs, layerParsePrms, 6);
```

Comment:
Why did you remove this check?

Reply:
MO provides LSTMCell with 6 inputs. Inside the LSTMCell op we are adding 7th input (peephole). After serialization LSTMCell IR format consists of 7 inputs and test is failing due to this check. Hardocded value in check is not correct. Besides this will be removed when LSTMCell will be deserialized.

Reply:
We shouldn't remove this check, because Serializer and Deserializer should be align with OpenVINO spec.
In OpenVINO spec [`LSTMCell`](https://docs.openvinotoolkit.org/latest/openvino_docs_ops_sequence_LSTMCell_1.html) has only 6 inputs.
@itikhono @lazarevevgeny Do you remember why we have this difference between specification and the nGraph operation?

Reply:
@itikhono , could you comment on that? I don't remember the whole story.

Reply:
LSTMCell v0 version has 7 inputs (one of these is the default input for peepholes, it is not used actually, there is a constructor of v0 version with 6 inputs to match the specification, default peepholes input adds in the definition of this constructor ), it's a mistake and it was fixed in LSTMCell v4 version in 2021.1 Release. 
I suppose there shouldn't be any problems with v4 version. The issue is relevant only for v0 version, I guess we need to have some workaround for that in serialization code (handle a special case for LSTMCell v0), maybe we should always serialize only 6 inputs for this version. @ilyachur @lazarevevgeny what do you think?

Reply:
Probably serializer could dump the v0 operation in a correct format with 6 inputs. Is it possible? Anyway, any working solution for the buggy v0 operation should be fine.

Reply:
WA on Serializer side provided.